diff --git a/tensorflow/lite/micro/tools/project_generation/create_tflm_tree.py b/tensorflow/lite/micro/tools/project_generation/create_tflm_tree.py
index e964a12c..fa058f8a 100644
--- a/tensorflow/lite/micro/tools/project_generation/create_tflm_tree.py
+++ b/tensorflow/lite/micro/tools/project_generation/create_tflm_tree.py
@@ -37,6 +37,11 @@ import shutil
 import subprocess
 
 
+# Needed to cleanup source files list when calling the script from CMake 
+def _check_file(x):
+  return os.path.exists(x) and os.path.isfile(x)
+  
+
 def _get_dirs(file_list):
   dirs = set()
   for filepath in file_list:
@@ -53,12 +58,12 @@ def _get_file_list(key, makefile_options, tensorflow_root):
                              stdout=subprocess.PIPE,
                              stderr=subprocess.PIPE)
   stdout, stderr = process.communicate()
-
+  
   if process.returncode != 0:
     raise RuntimeError("%s failed with \n\n %s" %
                        (" ".join(params_list), stderr.decode()))
 
-  return [bytepath.decode() for bytepath in stdout.split()]
+  return list(filter(_check_file, [bytepath.decode() for bytepath in stdout.split()]))
 
 
 def _third_party_src_and_dest_files(prefix_dir, makefile_options,
@@ -70,7 +75,6 @@ def _third_party_src_and_dest_files(prefix_dir, makefile_options,
   src_files.extend(
       _get_file_list("list_third_party_headers", makefile_options,
                      tensorflow_root))
-
   # The list_third_party_* rules give path relative to the root of the git repo.
   # However, in the output tree, we would like for the third_party code to be a
   # tree under prefix_dir/third_party, with the path to the tflm_download
